src/java.desktop/share/classes/java/awt/Polygon.java

```diff
@@ -103,7 +103,11 @@ public class Polygon implements Shape, java.io.Serializable {
 
     /**
      * The bounds of this {@code Polygon}.
-     * This value can be null.
+     * <p>
+     * When created, {@link #invalidate() invalidated}, or {@link #reset() reset},
+     * this becomes {@code null}. To get out of the null state,
+     * {@link #getBounds()} called with {@link #npoints} being greater than
+     * {@code 0} will transfer into a non-null {@link Rectangle}.
      *
```

Comment:
Another dubious piece of design. This should never have been exposed in the API.
What we need to say here is
/*
 * The value of this field is updated by the implementation which may set it to null and will
 *   re-evaluate it as needed. Application sub-classes should never access this field.
 */
And leave it at that.

Reply:
I noticed one thing: Phil suggested “re-evaluate”, you used “reevaluate” without the hyphen.

I tend to use the hyphened version; at the same time, articles in the internet suggest the hyphen is unneeded in such words.

I'm fine with either.